pkg/agent/flowexporter/exporter/exporter_perf_test.go

```diff
@@ -0,0 +1,269 @@
+	testPerfActiveFlowTimeout  = 2 * time.Second
+	testPerfIdleFlowTimeout    = 1 * time.Second
+)
+
+var count = 0
+var exp *flowExporter
+
+func BenchmarkExport(b *testing.B) {
+	disableLogToStderr()
+	ctrl := gomock.NewController(b)
+	defer ctrl.Finish()
+
+	registry := ipfix.NewIPFIXRegistry()
+	registry.LoadRegistry()
+	stopChan1 := make(chan struct{})
+	stopChan2 := make(chan struct{})
+	listener := startLocalServer(b, stopChan1, stopChan2)
+
+	conns := getConnections(testConnectionsCount)
+	records := flowrecords.NewFlowRecords()
+	mockIfaceStore := interfacestoretest.NewMockInterfaceStore(ctrl)
+	mockConnDumper := connectionstest.NewMockConnTrackDumper(ctrl)
+	mockProxier := proxytest.NewMockProxier(ctrl)
+	npQuerier := queriertest.NewMockAgentNetworkPolicyInfoQuerier(ctrl)
+	connStore := connections.NewConnectionStore(mockConnDumper, records, mockIfaceStore, true, false, mockProxier, npQuerier, testPollInterval)
+	var err error
+	exp, err = NewFlowExporter(connStore, records, listener.String(), listener.Network(), testPerfActiveFlowTimeout, testPerfIdleFlowTimeout, false, true, false, nil, nil, false)
+	if err != nil {
+		b.Errorf("error in starting flow exporter: %v", err)
+	}
+	// Generate connections for Dumpflows, each round with update connections, new connections and dying connections
+	mockConnDumper.EXPECT().DumpFlows(testZoneFilter).Return(conns, testConnectionsCount, nil).Times(1)
+	for i := 0; i < 5; i++ {
+		conns = getUpdateConnections(conns, testNewConnectionsCount, testUpdateConnectionsCount, testDyingConnectionsCount, i)
+		mockConnDumper.EXPECT().DumpFlows(testZoneFilter).Return(conns, len(conns), nil).Times(1)
+	}
+	mockConnDumper.EXPECT().DumpFlows(testZoneFilter).Return(conns, len(conns), nil).AnyTimes()
+	mockConnDumper.EXPECT().GetMaxConnections().Return(testConnectionsCount, nil).AnyTimes()
+	mockIfaceStore.EXPECT().GetInterfaceByIP(gomock.Any()).Return(nil, false).AnyTimes()
+
+	var maxAlloc uint64
+	go statMaxMemAlloc(&maxAlloc, 500*time.Millisecond, stopChan2)
+	go connStore.Run(stopChan1)
+	go exp.Run(stopChan2)
+	<-stopChan2
+	b.Logf(`Summary metrics:
+   TOTAL_CONNECTIONS	INIT_CONNECTIONS    UPDATE_CONNECTIONS/poll    NEW_CONNECTIONS/poll    DYING_CONNECTIONS/poll    POLL_INTERVAL(s)    MEMORY(M)
+%d	%d	%d	%d	%d	%d	%-12d 
+`, count, testConnectionsCount, testUpdateConnectionsCount, testNewConnectionsCount, testDyingConnectionsCount, testPollInterval/1000000, maxAlloc/1024/1024)
+}
+
+func startLocalServer(b *testing.B, stopChan1, stopChan2 chan struct{}) (address net.Addr) {
+	udpAddr, err := net.ResolveUDPAddr("udp", "127.0.0.1:0")
+	if err != nil {
+		b.Fatalf("Got error when resolving UDP address: %v", err)
+	}
+	conn, err := net.ListenUDP("udp", udpAddr)
+	if err != nil {
+		b.Fatalf("Got error when creating a local server: %v", err)
+	}
+	if err != nil {
+		b.Fatalf("Got error when creating a local server: %v", err)
+	}
+	prevCount := 0
+	ticker := time.NewTicker(testIdleTimeOut)
```

Comment:
use timer here?

Reply:
If we use timer instead of ticker, we will need to reset timer each time. Is there any special reason for timer?

Reply:
As we are sleeping for `testDuration`, I thought this ticker is only expected to trigger once as a timeout like the name suggests `testIdleTimeOut`. Looks like to update `prevCount`, you want this to be a periodic ticker. Could we just timeout once without worrying about how many messages are received?